File: perl/test/codegen.cc
#include <stdio.h>

template<bool bnValue>
void DoStupidPrint()
{
   if (bnValue) printf("Hallo Isch bin ein Berliner!");
}

int main()
{
   DoStupidPrint<true>();
   DoStupidPrint<false>();
   bool b = false;
   //DoStupidPrint<b>();
}

File: perl/test/codegen.pl
#!/usr/bin/perl -w

sub DoStupidPrint {
    my $bnValue = shift;
    eval qq[sub {
                if ($bnValue) {
                    print("Hallo Isch bin ein Berliner!\n");
                }
            }];
}


DoStupidPrint(0)->();
DoStupidPrint(1)->();

DoStupidPrint(someRuntimeValue())->();
